pallets/rating/src/lib.rs

```diff
@@ -0,0 +1,234 @@
+		/// A metedata entry has been added to the identifier.
+		/// \[rating identifier, controller\]
+		MetadataSet { identifier: IdentifierOf, controller: CordAccountOf<T> },
+		/// An identifier metadata entry has been cleared.
+		/// \[rating identifier, controller\]
+		MetadataCleared { identifier: IdentifierOf, controller: CordAccountOf<T> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Rating idenfier is not unique
+		RatingAlreadyAnchored,
+		/// Rating idenfier not found
+		RatingNotFound,
+		/// Only when the author is not the controller/delegate.
+		UnauthorizedOperation,
+		// Invalid creator signature
+		InvalidSignature,
+		//Rating hash is not unique
+		HashAlreadyAnchored,
+		// Expired Tx Signature
+		ExpiredSignature,
+		// Invalid Rating Identifier
+		InvalidRatingIdentifier,
+		// Invalid Schema Identifier Length
+		InvalidIdentifierLength,
+		//Rating digest is not unique
+		DigestHashAlreadyAnchored,
+		// Invalid transaction hash
+		InvalidTransactionHash,
+		// Metadata limit exceeded
+		MetadataLimitExceeded,
+		// Metadata already set for the entry
+		MetadataAlreadySet,
+		// Metadata not found for the entry
+		MetadataNotFound,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a new rating identifier and associates it with its
+		/// controller. The controller (issuer) is the owner of the identifier.
+		///
+		/// * origin: the identity of the Transaction Author. Transaction author
+		///   pays the transaction fees and the author identity can be different
+		///   from the rating issuer
+		/// * tx_rating: the incoming rating. Identifier is generated from the
+		///   genesis digest (hash) provided as part of the details
+		/// * tx_signature: signature of the issuer aganist the rating genesis
+		///   digest (hash).
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::create())]
+		pub fn create(
+			origin: OriginFor<T>,
+			tx_rating: RatingType<T>,
+			tx_signature: SignatureOf<T>,
+		) -> DispatchResult {
+			<T as Config>::EnsureOrigin::ensure_origin(origin)?;
+
+			ensure!(
+				!<RatingHashes<T>>::contains_key(&tx_rating.digest),
+				Error::<T>::InvalidTransactionHash
+			);
+
+			ensure!(
+				tx_signature.verify(&(&tx_rating.digest).encode()[..], &tx_rating.controller),
+				Error::<T>::InvalidSignature
+			);
+
+			let identifier: IdentifierOf = BoundedVec::<u8, ConstU32<48>>::try_from(
+				ss58identifier::generate(&(&tx_rating.digest).encode()[..], RATING_PREFIX)
+					.into_bytes(),
+			)
+			.map_err(|()| Error::<T>::InvalidIdentifierLength)?;
+
+			ensure!(!<Ratings<T>>::contains_key(&identifier), Error::<T>::RatingAlreadyAnchored);
+
+			<RatingHashes<T>>::insert(&tx_rating.digest, &identifier);
+
+			<Ratings<T>>::insert(
+				&identifier,
+				RatingDetails { rating: tx_rating.clone(), meta: false },
+			);
+
+			Self::deposit_event(Event::Anchor {
+				identifier,
+				digest: tx_rating.digest,
+				author: tx_rating.controller,
+			});
+
+			Ok(())
+		}
+	}
```

Comment:
Should we be having 'update()' method here? Depends on what updates would happen to rating.

Reply:
no update for now